src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -1063,10 +1062,9 @@ public virtual Task<IdentityResult> RemoveClaimAsync(TUser user, Claim claim)
                 throw new ArgumentNullException("roles");
             }
 
-            var userRoles = await userRoleStore.GetRolesAsync(user, CancellationToken);
             foreach (var role in roles)
```

Comment:
My guess is I was trying to avoid calling the store N times to do this check before, so this actually does fix and issue with AddRoles containing dupes itself, that was in the code before.  

Here's what IsInRole does, AnyAsync should pickup any prior updates correct?:

```
            var role = await Roles.SingleOrDefaultAsync(r => r.Name.ToUpper() == roleName.ToUpper(), cancellationToken);
            if (role != null)
            {
                var userId = user.Id;
                var roleId = role.Id;
                return await UserRoles.AnyAsync(ur => ur.RoleId.Equals(roleId) && ur.UserId.Equals(userId));
            }
```